src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.health = 100;
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.health = 100;
+  }
+
+  bite(prey) {
+    if (prey.hidden !== undefined && prey.hidden !== true) {
```

Comment:
```suggestion
    if (!prey.hidden) {
```

I guess it should work as well

Reply:
then subtracted health in carnivores